sdk/monitor/Azure.Monitor.Ingestion/tests/Samples/LogDataAndQueryAsync.cs

```diff
@@ -95,5 +98,127 @@ public async Task QueryDataAsync()
+            var ruleId = "<data_collection_rule_id>";
+            var streamName = "<stream_name>";
+
+#if SNIPPET
+            var credential = new DefaultAzureCredential();
+#else
+            TokenCredential credential = new DefaultAzureCredential();
+            endpoint = new Uri(TestEnvironment.DCREndpoint);
+            credential = TestEnvironment.Credential;
+#endif
+            LogsIngestionClient client = new(endpoint, credential);
+
+            DateTimeOffset currentTime = DateTimeOffset.UtcNow;
+
+            var entries = new List<Object>();
+            for (int i = 0; i < 100; i++)
+            {
+                entries.Add(new Object[] {
+                    new {
+                        Time = currentTime,
+                        Computer = "Computer" + i.ToString(),
+                        AdditionalContext = i
+                    }
+                });
+            }
+            // Set concurrency and EventHandler in UploadLogsOptions
+            UploadLogsOptions options = new UploadLogsOptions();
+            options.MaxConcurrency = 10;
+            options.UploadFailedEventHandler += Options_UploadFailed;
+
+            // Upload our logs
+            Response response = await client.UploadAsync(ruleId, streamName, entries, options).ConfigureAwait(false);
+
+            Task Options_UploadFailed(UploadFailedEventArgs e)
+            {
+                // Throw exception from EventHandler to stop Upload if there is a failure
+                IReadOnlyList<object> failedLogs = e.FailedLogs;
+                throw e.Exception;
+            }
+            #endregion
+        }
+
+        public async Task LogDataIEnumerableAsync()
+        {
+            #region Snippet:UploadLogDataIEnumerableAsync
+            var endpoint = new Uri("<data_collection_endpoint_uri>");
+            var ruleId = "<data_collection_rule_id>";
+            var streamName = "<stream_name>";
+
+#if SNIPPET
+            var credential = new DefaultAzureCredential();
+#else
+            TokenCredential credential = new DefaultAzureCredential();
+            endpoint = new Uri(TestEnvironment.DCREndpoint);
+            credential = TestEnvironment.Credential;
+#endif
+            LogsIngestionClient client = new(endpoint, credential);
+
+            DateTimeOffset currentTime = DateTimeOffset.UtcNow;
+
+            var entries = new List<Object>();
+            for (int i = 0; i < 100; i++)
+            {
+                entries.Add(new Object[] {
+                    new {
+                        Time = currentTime,
+                        Computer = "Computer" + i.ToString(),
+                        AdditionalContext = i
+                    }
+                });
+            }
+
+            // Upload our logs
+            Response response = await client.UploadAsync(ruleId, streamName, entries).ConfigureAwait(false);
+            #endregion
+        }
+
+        public async Task LogDataIEnumerableCancellationTokenAsync()
```

Comment:
I thought we weren't going to show this one in a sample. I would move this to a test instead.